### PR TITLE
#163518655 Add following functionality on readArticle page

### PR DESCRIPTION
**Pull request by Musigwa-zz**
#### What does this PR do?

- It adds the ability to follow an author
#### Description of Task to be completed?

- Added the following action to the `follow` button
- Implemented the functionality of clicking twice to unfollow

#### How should this be manually tested?
- After cloning the repository, check out `ft-follow-another-user-163518655` branch
- Run `yarn` to install the project dependencies
- Run `yarn start` to start the application then, you navigate to any of the articles
- There you shall see the follow button next to the author's info of that article
- If you're already following that author, click to unfollow and vice-versa

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- #163518655

### CODE REVIEW

**Comment by oesukam** [comment]
@Musigwa Which format are you using for the PR's title?

File: src/views/ReadArticle.js [hunk 1]
```diff
@@ -252,6 +272,7 @@ export class Article extends Component {
                       className="btn delete_article"
                       onClick={() => this.handleDeleteArticle()}
```
Comment (niomwungeri-fabrice) [inline src/views/ReadArticle.js]: What do you think using references instead of a function call as we agreed in our team?

**Review by niomwungeri-fabrice — APPROVED** [review]
Good Job!

**Review by abayo-luc — CHANGES_REQUESTED** [review]
Well done on the part of the codebase. It looks clean. However, I tried to follow a user and unfollow them immediately, and it was not responding. Check if there is some improvement need to be done. 
Thanks!

File: src/redux/reducers/followingReducer.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  WAITING_RESPONSE
+} from "../actionTypes";
+
+export const initialState = {
+  isFetching: false,
+  status: null
+};
+
+export default (state = initialState, action) => {
+  const { type, payload } = action;
```
Comment (oesukam) [inline src/redux/reducers/followingReducer.js]: Should be destructured on line 12.

File: src/views/ReadArticle.js [hunk 3]
```diff
@@ -27,11 +29,13 @@ import bookmarkIcon from "../assets/img/bookmark-icons.svg";
 export const mapStateToProps = state => ({
   currentUser: state.login.currentUser,
   asideArticles: state.fetchedArticle.asideArticles,
-  article: state.fetchedArticle
+  article: state.fetchedArticle,
+  following: state.following
 });
```
Comment (oesukam) [inline src/views/ReadArticle.js:40]: Empty line required

File: src/views/ReadArticle.js [hunk 4]
```diff
@@ -200,10 +210,11 @@ export class Article extends Component {
         readTime,
         comments
       } = retrievedArticle);
-      ({ username, firstName, lastName, image } = author);
+      ({ username, firstName, following, lastName, image } = author);
     }
     const isAuthor = isCurrentUserAuthor(username, currentUser);
-
+    const { following: followObject } = this.props;
+    if (followObject.status !== null) following = followObject.status;
```
Comment (oesukam) [inline src/views/ReadArticle.js]: What happens if status === undefined? Can't you just check for `if(followObject.status)`?

File: src/views/ReadArticle.js (hunk 1 above)
Comment (oesukam) [inline src/views/ReadArticle.js]: Change to function's reference

Comment (oesukam) [inline src/views/ReadArticle.js]: Function's reference

**Review by oesukam — CHANGES_REQUESTED** [review]

File: src/views/ReadArticle.js (hunk 4 above)
Comment (Musigwa-zz) [reply src/views/ReadArticle.js]: I see, but what if I check for `if(followObject.status)` and status becomes `false` since it is a boolean?

**Comment by Musigwa-zz** [comment]
> Well done on the part of the codebase. It looks clean. However, I tried to follow a user and unfollow them immediately, and it was not responding. Check if there is some improvement need to be done.
> Thanks!

Okay, for the attentive testing. I'll look at it and solve the issue.

File: src/redux/reducers/followingReducer.js (hunk 2 above)
Comment (Musigwa-zz) [reply src/redux/reducers/followingReducer.js]: Yeah sure.

**Review by abayo-luc — APPROVED** [review]

**Review by oesukam — APPROVED** [review]

**Pull request merged**